extensions/dicom-seg/src/OHIFDicomSegSopClassHandler.js

```diff
@@ -0,0 +1,68 @@
+import { MODULE_TYPES, utils } from '@ohif/core';
+import loadSegmentation from './loadSegmentation';
+
+import id from './id';
+
+// TODO: Should probably use dcmjs for this
+const SOP_CLASS_UIDS = {
+  DICOM_SEG: '1.2.840.10008.5.1.4.1.1.66.4',
+};
+
+const sopClassUids = Object.values(SOP_CLASS_UIDS);
+
+// TODO: Handle the case where there is more than one SOP Class Handler for the
+// same SOP Class.
+const OHIFDicomSegSopClassHandler = {
+  id: 'OHIFDicomSegSopClassHandler',
+  type: MODULE_TYPES.SOP_CLASS_HANDLER,
+  sopClassUids,
+  getDisplaySetFromSeries: function(
+    series,
+    study,
+    dicomWebClient,
+    authorizationHeaders
+  ) {
+    const instance = series.getFirstInstance();
+    const referencedSeriesSequence = instance.getTagValue(
+      'ReferencedSeriesSequence'
+    );
+    const frameOfReferenceUID = instance.getTagValue('FrameOfReferenceUID');
+    const { seriesDate, seriesTime, seriesDescription } = series.getData();
+
+    const segDisplaySet = {
+      plugin: id,
+      modality: 'SEG',
+      displaySetInstanceUid: utils.guid(),
+      wadoRoot: study.getData().wadoRoot,
+      wadoUri: instance.getData().wadouri,
+      sopInstanceUid: instance.getSOPInstanceUID(),
+      seriesInstanceUid: series.getSeriesInstanceUID(),
+      studyInstanceUid: study.getStudyInstanceUID(),
+      referencedSeriesSequence,
+      frameOfReferenceUID,
+      authorizationHeaders,
+      isDerived: true,
+      referencedDisplaySetUid: null, // Assigned when loaded.
+      labelmapIndex: null, // Assigned when loaded.
+      isLoaded: false,
+      seriesDate,
+      seriesTime,
+      seriesDescription,
+    };
+
+    segDisplaySet.load = function(referencedDisplaySet, studies) {
```

Comment:
@dannyrb , are you ok with this hitting master? Its a bit of a funny pattern. If you guys can think of a better way to 

What we need is to have a custom load process that can be triggered on demand when you search for a display set, I can't think of a better way to do this off the top of my head. Are we happy with `load`, as I guess this is becoming cemented as api?

Reply:
I guess. Before merge, I'd like to see the `sopClassHandler` docs updated to indicate that this optional/convention. This is something you, @swederik, and I should think on for our architecture talks (now that we're slowly deprecating `sopClassHandler` in extensions)

Reply:
@dannyrb this is pending?